crates/config/src/consensus/mod.rs

```diff
@@ -0,0 +1,50 @@
+use std::borrow::Cow;
+
+use serde::{Deserialize, Serialize};
+
+use crate::ForkConfig;
+
+mod builtins;
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub enum BuiltinConsensus {
+    Mainnet,
+    Testnet,
+}
+
+/// Represents a consensus config
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged, deny_unknown_fields)]
+pub enum Consensus {
+    /// Builtin consensus config
+    Builtin {
+        /// The identifier of the builtin consensus config.
+        builtin: BuiltinConsensus,
+    },
+    /// Customized config
+    Config {
+        /// The config
+        config: Box<ForkConfig>,
```

Comment:
Why boxing here?

Reply:
To remove the clips warning large_enum_variant.

Reply:
If we need to box it, maybe use `Arc` to make clone cheap.

Reply:
Arc does not implement serde; we need to do it manually. 
We rarely use `ForkConfig#clone` during the node running, so IMO, Box is enough.